src/stores/user/actions/index.ts

```diff
@@ -55,8 +51,7 @@ export const getUserDetailsByAddressFromContract = async (address: string): Prom
 }
 
 export const saveUserDetails = async (details: Partial<User>, address: string): Promise<{ result: User }> => {
-  const userSigner = new InjectedArweaveSigner(window.arweaveWallet)
-  await userSigner.setPublicKey()
+  const userSigner = await getSigner()
   userSigner.getAddress = () => userSigner.signer.getActiveAddress()
```

Comment:
othent signer doesn't have `getActiveAddress` but has `getActiveKey` so could you please replace this line with:

```
userSigner.getAddress = () => useGlobalStore.getState().authState.address
```
OR,

This line is not needed at all if we use contract without the signer with the viewState.

Reply:
this has been updated now